rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/WorkerBookKeeper.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2016. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+
+package com.qubole.rubix.bookkeeper;
+
+import com.codahale.metrics.MetricRegistry;
+import com.qubole.rubix.bookkeeper.manager.WorkerManager;
+import org.apache.hadoop.conf.Configuration;
+
+import java.io.FileNotFoundException;
+
+public class WorkerBookKeeper extends BookKeeper
+{
+  // The manager used when running on a worker node.
+  private final WorkerManager workerManager;
+
+  public WorkerBookKeeper(Configuration conf, MetricRegistry metrics, WorkerManager workerManager) throws FileNotFoundException
+  {
+    super(conf, metrics);
+    this.workerManager = workerManager;
+  }
+
+  @Override
+  public void handleHeartbeat(String workerHostname)
```

Comment:
Is this required ? I dont see an equivalent abstract method in `Bookkeeper`.

Reply:
This overrides `handleHeartbeat` from BookKeeperService.Iface, which `BookKeeper` implements.

Reply:
Oh - `Iface` is created from thrift spec ?

Reply:
We generate BookKeeperService.Iface everytime we build the jars and that file is in the target directory. I also had the same confusion.